compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -374,12 +361,30 @@ class ROCMTargetBackend final : public TargetBackend {
         llvmFunc->addFnAttr(
             "amdgpu-flat-work-group-size",
             (llvm::Twine("1, ") + llvm::Twine(flatWgSize)).str());
-        if (wavesPerEu > 0) {
-          llvmFunc->addFnAttr("amdgpu-waves-per-eu",
-                              std::to_string(wavesPerEu));
-        }
         if (targetArch.starts_with("gfx9"))
           addPreloadKernArgHint(llvmFunc);
+
+        // Set the amdgpu-waves-per-eu flag from config if given.
+        if (std::optional<IntegerAttr> attr =
+                getConfigIntegerAttr(targetAttr, "waves_per_eu")) {
```

Comment:
Do we still need this config method of setting it?

Reply:
+1, I'd expect waves-per-eu to use the new mechanism

Reply:
I plan to generalize the global config setting flag for llvm attrs in another pr. This pr generalizes this for translation_info attrs only.